scripts/update_e2e_coverage.py

```diff
@@ -0,0 +1,180 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Updates coverage of TensorFlow e2e tests on all backends.
+
+Example usage: python3 update_e2e_coverage.py build-docs
+"""
+
+import argparse
+import collections
+import os
+import subprocess
+
+REFERENCE_BACKEND = 'tf'
+# Assumes that tests are expanded for the tf_also, iree_vmla, iree_llvmjit and
+# iree_vulkan backends.
+BACKENDS_TO_TITLES = collections.OrderedDict([
+    ('tf_also', 'tensorflow'),
+    ('iree_vmla', 'vmla'),
+    ('iree_llvmjit', 'llvm-ir'),
+    ('iree_vulkan', 'vulkan-spirv'),
+])
+
+TEST_SUITES_TO_HEADERS = {
+    '//integrations/tensorflow/e2e:e2e_tests':
+        'End to end TensorFlow tests',
+    '//integrations/tensorflow/e2e/keras:keras_tests':
+        'End to end tests written using tf.keras',
+    '//integrations/tensorflow/e2e/keras:vision_external_tests':
+        'End to end tests of tf.keras.applications vision models',
+}
+
+# Some test suites are generated from a single source. This allows us to point
+# to the right test file when generating test URLs.
+SINGLE_SOURCE_SUITES = {
+    '//integrations/tensorflow/e2e/keras:vision_external_tests':
+        'vision_model_test',
+}
+
+# The symbols to show in the table if the operation is supported or not.
+SUCCESS_ELEMENT = '<span class="success-table-element">✓</span>'
+FAILURE_ELEMENT = '<span class="failure-table-element">✗</span>'
+
+MAIN_URL = 'https://github.com/google/iree/tree/main'
+TARGETS_URL = os.path.join(MAIN_URL, 'iree/compiler/Dialect/HAL/Target')
+
+E2E_COVERAGE_DESCRIPTION = f"""# TensorFlow End to End Coverage
+There are three backend [targets]({TARGETS_URL}) in IREE:
+
+- vmla
+- llvm-ir
+- vulkan-spirv
+
+The table shows the supported TensorFlow functions and models on each backend.
+
+"""
+
+
+def parse_arguments():
+  """Parses command-line options."""
+  parser = argparse.ArgumentParser(
+      description='Generates Markdown files for op coverage table')
+  parser.add_argument(
+      'build_dir', metavar='BUILD_PATH', type=str, help='Base build directory.')
+
+  parsed_args = parser.parse_args()
+  if not os.path.isdir(parsed_args.build_dir):
+    raise parser.error('expected path to a directory')
+
+  return parsed_args
+
+
+def create_markdown_table(rows):
+  """Converts a 2D array to a Markdown table."""
+  return '\n'.join([' | '.join(row) for row in rows])
+
+
+def get_name_and_backend(test_string):
+  """Splits a pathless test target into its name and comparison backend."""
+  name, backend = test_string.split(f'__{REFERENCE_BACKEND}__')
+  return name, backend
+
+
+def get_test_targets(test_suite_path):
+  """Returns a list of test targets stripped of paths and suite names."""
+  # Check if the suite exists (which may not be true for _failing suites)
```

Comment:
Why a `_` before `failing` ?

Reply:
I included that subconsciously since we generate the failing targets to check by appending `_failing` to the passing ones, but that doesn't need to be in this comment.